src/server/semester/semester.service.ts

```diff
@@ -59,4 +80,188 @@ export class SemesterService {
+          meetings: [],
+        }));
+
+      // Create new fall semester
+      const fallSemester = new Semester();
+      // NOTE: academic year in Semester entity is actually the calendar year
+      fallSemester.academicYear = (newAcademicYear - 1).toString();
+      fallSemester.term = TERM.FALL;
+      fallSemester.courseInstances = newFallInstances;
+
+      this.logService.verbose(`Created ${fallSemester.courseInstances.length} fall course instances.`);
+      this.logService.debug(fallSemester.courseInstances);
+
+      this.logService.verbose('Creating the fall non-class events.');
+
+      fallSemester.nonClassEvents = existingSpringNonClassEvents
+        .map((nce) => ({
+          ...new NonClassEvent(),
+          nonClassParent: nce.nonClassParent,
+          private: nce.private,
+          meetings: [],
+        }));
+
+      this.logService.verbose(`Created ${fallSemester.nonClassEvents.length} fall non-class events.`);
+      this.logService.debug(fallSemester.nonClassEvents);
+
+      this.logService.verbose('Creating the fall absences.');
+
+      fallSemester.absences = existingSpringAbsences
+        .map((absence) => ({
+          ...new Absence(),
+          faculty: absence.faculty,
+        }));
+
+      this.logService.verbose(`Created ${fallSemester.absences.length} fall absences.`);
+      this.logService.debug(fallSemester.absences);
+
+      this.logService.verbose('Creating the spring course instances.');
+      this.logService.verbose('Creating the spring non-class events.');
+      this.logService.verbose('Creating the spring absences.');
+
+      // Create new spring semester
+      const springSemester = {
+        ...fallSemester,
+        courseInstances: fallSemester.courseInstances.map(
+          (instance) => ({ ...instance })
+        ),
+        nonClassEvents: fallSemester.nonClassEvents.map(
+          (event) => ({ ...event })
+        ),
+        absences: fallSemester.courseInstances.map(
+          (absence) => ({ ...absence })
+        ),
+        academicYear: newAcademicYear.toString(),
+        term: TERM.SPRING,
+      };
+
+      this.logService.verbose(`Created ${springSemester.courseInstances.length} spring course instances.`);
+      this.logService.debug(springSemester.courseInstances);
+      this.logService.verbose(`Created ${springSemester.nonClassEvents.length} spring non-class events.`);
+      this.logService.debug(springSemester.nonClassEvents);
+      this.logService.verbose(`Created ${springSemester.absences.length} spring absences.`);
+      this.logService.debug(springSemester.absences);
+
+      // Note that this also saves the nested entities because of
+      // the cascade set on semester
+      await this.semesterRepository.save([fallSemester, springSemester]);
+
+      this.logService.info(`Successfully created academic year ${newAcademicYear}.`);
+    }
+    this.logService.info(`Academic year ${newAcademicYear} already exists.`);
```

Comment:
Ah! I don't know how I missed this. This was fixed in [7469279](https://github.com/seas-computing/course-planner/pull/496/commits/74692794e2608eed25cb9d9a25bcefa1ac507dce) so that the "already exists" message doesn't print out after the year is successfully created.